src/main/jbake/content/hawkular-apm/index.adoc

```diff
@@ -0,0 +1,124 @@
+= Hawkular APM
+Gary Brown
+2016-11-07
+:description: OpenTracing compliant application performance management
+:jbake-type: page
+:jbake-status: published
+:icons: font
+:toc: macro
+:toc-title:
+
+toc::[]
+
+
+== What is Application Performance Management?
+
+As defined on https://en.wikipedia.org/wiki/Application_performance_management[wikipedia], _"application performance management (APM) is the monitoring and management of performance and availability of software applications. APM strives to detect and diagnose complex application performance problems to maintain an expected level of service"_.
+
+In a microservices world this becomes even more critical due to the potentially large number of simple services that collaborate to deliver a business application.
+
+Being able to obtain high level aggregated views of an application's performance, as well as trace the individual path taken by a particular end user request through a potentially dynamically scaled cloud environment, enables application operations and development teams to understand where issues may have occurred and what improvements can be made to deliver better performance for the business.
+
+
+== Features
+
+This section provides a high level description of the features available in Hawkular APM. For more detailed information, including how install and use the project, please see the https://hawkular.gitbooks.io/hawkular-apm-user-guide/content/[user guide].
+
+=== Application Instrumentation
+
+Hawkular APM offers three approaches for instrumenting your application.
+
+==== OpenTracing
+
+http://opentracing.io/[OpenTracing] is a _"vendor-neutral open standard for distributed tracing"_. The standard currently provides APIs and abstract implementations for 6 languages, _Java_, _Javascript_, _Go_, _Python_, _Objective C_ and _C++_.
+
+The Hawkular APM project currently provides the following OpenTracing compliant providers:
+
+.OpenTracing Providers for Hawkular APM
+|===
+|Language |Code |Available
+|Java |https://github.com/hawkular/hawkular-apm/tree/master/client/opentracing |0.11
+|Javascript |https://github.com/hawkular/hawkular-apm-opentracing-javascript |Scheduled for 0.12
+|===
+
+Note: It is also possible to use OpenTracing compliant zipkin providers. However Hawkular APM and Zipkin providers cannot be mixed within the same environment (i.e. propagation of state is not compatible between the two projects).
+
+==== JVM based agent
+
+A java agent can be configured to non-intrusively monitor your applications/services and report trace information to the APM server.
+
+==== Zipkin compliant client libraries
```

Comment:
> Zipkin is a distributed tracing system. If your application already makes use of its client libraries to capture trace information, you can benefit from our drop-in replacement for the server: just deploy Hawkular APM and point your client libraries to our server and it will just work.